src/rpdk/core/data/managed-upload-infrastructure.yaml

```
@@ -3,6 +3,14 @@ Description: >
   This CloudFormation template provisions all the infrastructure that is
   required to upload artifacts to CloudFormation's managed experience.
 
+Parameters:
+  DestinationBucketName:
+    Description: Set this if you want to write access logs to a bucket you manage
+    Type: String
+    Default: ""
```

```suggestion
  DestinationBucketName:
    Description: Set this if you want to write access logs to a bucket you manage
    Type: String
    Default: ""
    AllowedPattern: "^[a-z0-9.-]*$"
```

We can't be as strict as the [AWS::S3::Bucket.BucketName AllowedPattern](https://github.com/aws-cloudformation/cfn-python-lint/pull/1208) because of the empty default, but we can at least constrain with some [AllowedPattern](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties) to prevent invalid S3 bucket name characters

---

Quick AllowedPattern note:`^[a-z0-9][a-z0-9.-]*[a-z0-9]$|^$` should catch both cases

---

please check https://github.com/aws-cloudformation/cloudformation-cli/pull/396#issuecomment-598676122